### DISCUSSION TITLE
Script - Readings MCP3021 ADC values via i2c

### DISCUSSION THREAD

**Discussion by letshin** [arendst/Tasmota#20550] [Q&A] [answered]
Sorry - I'm probably getting a lot of this wrong, but from my understanding of the [datasheet](https://ww1.microchip.com/downloads/aemDocuments/documents/APID/ProductDocuments/DataSheets/20001805C.pdf) I've done this:

When reading a single MCP3021 with either addresses `0x4d or 0x4a`, reading ir2(address) returns the ADC readings, but with two MCP3021 sensors attached no voltage reading is obtained doesn't work. The current code that works is:

Edit: In the examples below I did not shift the receiving 2 bytes by two bits to the right to account for the empty last two bits. 

With both devices connected, this works for a single device.
```
>D
found=0
val0=0

>B
; check device
found=ia(0x4d)

; Run every second
>S
 if found>0 {
  ;3v at 4095. Because of 3v v_ref in electronics
  val0=ir2(0x4d)/4095*3
  print %val0%
 }
```

With both devices connected, both devices cannot be read with the example below, they show the same values so there is some cross talk going on?
```
>D
found0=0
found1=0
val0=0
val1=0

>B
; check device
found0=ia(0x4d)
found1=ia(0x4a)

; Run every second
>S
 if found0>0 {
  ;3v at 4095. Because of 3v v_ref in electronics
  val0=ir2(0x4d)/4095*3
  print %val0%
 }
 if found1>0 {
  ;3v at 4095. Because of 3v v_ref in electronics
  val1=ir2(0x4a)/4095*3
  print %val1%
 }
```

What am I doing wrong please?

**Comment by gemu2015** [comment, marked as answer]
all scripter i2c cmds relate to one "opened" device, if you want to query multiple device you have to repeat the ia(x) cmd before each access.

This should work

```
>D
found0=0
found1=0
val0=0
val1=0

>B
; check device
found0=ia(0x4d)
found1=ia(0x4a)

; Run every second
>S
 if found0>0 {
  ;3v at 4095. Because of 3v v_ref in electronics
  ia(0x4d)
  val0=ir2(0x4d)/4095*3
  print %val0%
 }
 if found1>0 {
  ;3v at 4095. Because of 3v v_ref in electronics
  ia(0x4a)
  val1=ir2(0x4a)/4095*3
  print %val1%
 }
```

**Reply by letshin** [reply]
Thanks, this works!
